src/headless/HeadlessTools.cpp

```diff
@@ -0,0 +1,29 @@
+#include "HeadlessApps.h"
+#include "HeadlessPluginLayerProxy.h"
+
+namespace Surge
+{
+namespace Headless
+{
+
+SurgeSynthesizer* createSurge()
+{
+   HeadlessPluginLayerProxy* parent = new HeadlessPluginLayerProxy();
+   SurgeSynthesizer* surge = new SurgeSynthesizer(parent);
+   surge->setSamplerate(44100);
+   return surge;
+}
+
+bool loadPatchByIndex(SurgeSynthesizer* s, int index)
+{
+   if (index > s->storage.patch_list.size() || index < 0)
+      return false;
+
+   s->patchid_queue = index;
+   s->processThreadunsafeOperations();
```

Comment:
I looked in call sites and I think we could reorganize the code in a way that there would instead a function:

``
void SurgeSynthesizer::loadPatch(int id)
{
   s->patchid_queue = index;
   s->processThreadunsafeOperations(); 
}
```

I think it would be a good idea to abstract away the exact load mechanism.

Reply:
Yeah I agree. We can’t call it loadPatch I don’t think since SurgeSynthesizer already has one of those which does the innards, but we can call it something reasonable. “setPatchIndex” or something

Reply:
How do oadPatchInBackgroundThread and processThreadunsafeOperations() collaborate? The existing loadPatch() could loadPatchPrivate().

Reply:
yeah that’s the right thing to do. I haven’t figured out how the UI thread and audio thread and headless all collaborate in detail so I may be able to call the direct method here also. This needs resolving for sure.

Reply:
Turns out we don't need this at all; since there is no "UI Thread" in headless we can just call loadPatch directly.